extra/liquidexport

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env python3
+"""liquidexport – host a metrics HTTP endpoint with Prometheus formatted data from liquidctl
+
+This is an experimental script that collects stats from liquidctl and exposes them as a http://localhost:8098/metrics
+endpoint in the Prometheus text format.
+See: https://prometheus.io/docs/instrumenting/exposition_formats/#text-format-example
+
+Example metric with labels:
+# HELP liquidctl liquidctl exported metrics
+# TYPE liquidctl gauge
+liquidctl{device="NZXT Kraken X (X42, X52, X62 or X72)",sensor="liquid_temperature",unit="°C"} 33.6
+
+Usage:
+  liquidexport [options]
+  liquidexport --help
+  liquidexport --version
+
+Options:
+  --legacy-690lc          Use Asetek 690LC in legacy mode (old Krakens)
+  --server-port <number>  Port for the HTTP /metrics endpoint
+  -v, --verbose           Output additional information
+  -g, --debug             Show debug information on stderr
+  --version               Display the version number
+  --help                  Show this message
+
+Examples:
+  liquidexport
+  liquidexport --product 0xb200
+  liquidexport --interval 0.5
+
+Copyright (C) 2020  Alex Berryman
+SPDX-License-Identifier: GPL-3.0-or-later
+"""
+
+import logging
+import sys
+import time
+import usb
+
+from docopt import docopt
+from liquidctl.driver import *
+from prometheus_client import start_http_server
+from prometheus_client.core import GaugeMetricFamily, REGISTRY, InfoMetricFamily
+from datetime import timedelta
+
+LOGGER = logging.getLogger(__name__)
+
+
+def gauge_name_sanitize(name):
+    return name.replace(" ", "_").lower()
+
+
+class LiquidCollector(object):
+    def __init__(self):
+        self.description = 'liquidctl exported metrics'
+
+    def collect(self):
+        labels = ['device', 'sensor', 'unit']
+        g = GaugeMetricFamily('liquidctl', self.description, labels=labels)
+        i = InfoMetricFamily('liquidctl', self.description, labels=['device'])
+        for d in devs:
+            try:
+                get_status = d.get_status()
+                for metric in get_status:
+                    sanitized_name = gauge_name_sanitize(metric[0])
+                    sample_value = metric[1]
+                    unit = metric[2]
+
+                    if isinstance(sample_value, timedelta):
+                        # cast timedelta into seconds and override the supplied unit
+                        sample_value = sample_value.seconds
+                        unit = 'seconds'
+
+                    if unit != '':
+                        label_values = [d.description, sanitized_name, unit]
+                        g.add_metric(label_values, value=sample_value)
+                        LOGGER.debug(
+                            '%s: %s as GaugeMetric %s labels %s' %
+                            (d.description, metric, sanitized_name, '/'.join(label_values)))
```

Comment:
You can also let the `logging` module apply the `%` operator for you, which is what results in the string being constructed lazily.

```suggestion
                        LOGGER.debug(
                            '%s: %s as GaugeMetric %s labels %s',
                            d.description, metric, sanitized_name, '/'.join(label_values))
```